Kernel/Syscalls/mmap.cpp

```diff
@@ -417,4 +417,46 @@ int Process::sys$munmap(void* addr, size_t size)
     return -EINVAL;
 }
 
+void* Process::sys$allocate_tls(size_t size)
+{
+    REQUIRE_PROMISE(stdio);
+
+    dbg() << "allocate TLS: " << size;
```

Comment:
This syscall is clever, way better of a solution than I was thinking. I think other systems do the entirety of TLS allocation in userspace, but they don't have the privelidge of starting from scratch. Worth some thought whether doing full userspace only is worth it. One thing that will be annoying in the future is that for C++11 `thread_local`, GCC emits calls to `___tls_getaddr` I think. So, needs at least some help from the loader/libc to get the correct TLS location.